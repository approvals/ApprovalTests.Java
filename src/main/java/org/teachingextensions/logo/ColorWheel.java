package org.teachingextensions.logo;

import java.awt.Color;

/**
 * ColorWheel is a place to store a color palette. <br/>
 * <b>Example:</b> If you have a palette of
 * <font color="blue">blue</font>,
 * <font color="red">red</font>,
 * and <font color="purple">purple</font> <br/>
 * and then used it to print out the numbers, you would get <br/>
 * <font color="blue">1</font>
 * <font color="red">2</font>
 * <font color="purple">3</font>
 * <font color="blue">4</font>
 * <font color="red">5</font>
 * <font color="purple">6</font>
 * <font color="blue">7</font>
 * <font color="red">8</font>
 * <font color="purple">9</font>
 */
public class ColorWheel
{
  private static Wheel<Color> wheel = new Wheel<Color>();
  /**
   * This method adds a color to the ColorWheel. <br/>
   * <b>Example:</b> {@code  ColorWheel.addColor(Colors.Reds.Red);}
   * 
   * @param color
   *          the color to add to the wheel
   */
  public static void addColor(Color color)
  {
    wheel.add(color);
  }
  /**
   * This method returns the next color of the ColorWheel. <br/>
   * <b>Example:</b> {@code  Color penColor = ColorWheel.getNextColor();}
   * 
   * @return the next color of the ColorWheel
   */
  public static Color getNextColor()
  {
    return wheel.next();
  }
  /**
   * This method returns the a random color from the options on the ColorWheel. <br/>
   * <b>Example:</b> {@code  Color penColor = ColorWheel.getNextColor();}
   * 
   * @return A random color from the ColorWheel
   */
  public static Color getRandomColorFromWheel()
  {
    return wheel.getRandomFrom();
  }
  public static void removeAllColors()
  {
    wheel.empty();
  }
}
