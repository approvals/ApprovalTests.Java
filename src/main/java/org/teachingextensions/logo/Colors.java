package org.teachingextensions.logo;

import java.awt.Color;
import java.util.Random;

/**
 * Every color under the
 * <font color="red">r</font><font color="yellow">a</font><font color="orange">i</font><font color="green">n</font><font color="blue">b</font><font color="purple">o</font><font
 * color="blue">w</font>,
 * and different ways to get them.
 */
public class Colors
{
  public static class Blues
  {
    /**
     * The color <font color="#000080"> Navy </font>
     **/
    public static final Color Navy            = Color.decode("#000080");
    /**
     * The color <font color="#00008B"> DarkBlue </font>
     **/
    public static final Color DarkBlue        = Color.decode("#00008B");
    /**
     * The color <font color="#191970"> MidnightBlue </font>
     **/
    public static final Color MidnightBlue    = Color.decode("#191970");
    /**
     * The color <font color="#0000CD"> MediumBlue </font>
     **/
    public static final Color MediumBlue      = Color.decode("#0000CD");
    /**
     * The color <font color="#0000FF"> Blue </font>
     **/
    public static final Color Blue            = Color.decode("#0000FF");
    /**
     * The color <font color="#008080"> Teal </font>
     **/
    public static final Color Teal            = Color.decode("#008080");
    /**
     * The color <font color="#483D8B"> DarkSlateBlue </font>
     **/
    public static final Color DarkSlateBlue   = Color.decode("#483D8B");
    /**
     * The color <font color="#008B8B"> DarkCyan </font>
     **/
    public static final Color DarkCyan        = Color.decode("#008B8B");
    /**
     * The color <font color="#4682B4"> SteelBlue </font>
     **/
    public static final Color SteelBlue       = Color.decode("#4682B4");
    /**
     * The color <font color="#4169E1"> RoyalBlue </font>
     **/
    public static final Color RoyalBlue       = Color.decode("#4169E1");
    /**
     * The color <font color="#6A5ACD"> SlateBlue </font>
     **/
    public static final Color SlateBlue       = Color.decode("#6A5ACD");
    /**
     * The color <font color="#5F9EA0"> CadetBlue </font>
     **/
    public static final Color CadetBlue       = Color.decode("#5F9EA0");
    /**
     * The color <font color="#00CED1"> DarkTurquoise </font>
     **/
    public static final Color DarkTurquoise   = Color.decode("#00CED1");
    /**
     * The color <font color="#1E90FF"> DodgerBlue </font>
     **/
    public static final Color DodgerBlue      = Color.decode("#1E90FF");
    /**
     * The color <font color="#00BFFF"> DeepSkyBlue </font>
     **/
    public static final Color DeepSkyBlue     = Color.decode("#00BFFF");
    /**
     * The color <font color="#7B68EE"> MediumSlateBlue </font>
     **/
    public static final Color MediumSlateBlue = Color.decode("#7B68EE");
    /**
     * The color <font color="#48D1CC"> MediumTurquoise </font>
     **/
    public static final Color MediumTurquoise = Color.decode("#48D1CC");
    /**
     * The color <font color="#6495ED"> CornflowerBlue </font>
     **/
    public static final Color CornflowerBlue  = Color.decode("#6495ED");
    /**
     * The color <font color="#40E0D0"> Turquoise </font>
     **/
    public static final Color Turquoise       = Color.decode("#40E0D0");
    /**
     * The color <font color="#00FFFF"> Aqua </font>
     **/
    public static final Color Aqua            = Color.decode("#00FFFF");
    /**
     * The color <font color="#00FFFF"> Cyan </font>
     **/
    public static final Color Cyan            = Color.decode("#00FFFF");
    /**
     * The color <font color="#87CEEB"> SkyBlue </font>
     **/
    public static final Color SkyBlue         = Color.decode("#87CEEB");
    /**
     * The color <font color="#87CEFA"> LightSkyBlue </font>
     **/
    public static final Color LightSkyBlue    = Color.decode("#87CEFA");
    /**
     * The color <font color="#7FFFD4"> Aquamarine </font>
     **/
    public static final Color Aquamarine      = Color.decode("#7FFFD4");
    /**
     * The color <font color="#B0C4DE"> LightSteelBlue </font>
     **/
    public static final Color LightSteelBlue  = Color.decode("#B0C4DE");
    /**
     * The color <font color="#ADD8E6"> LightBlue </font>
     **/
    public static final Color LightBlue       = Color.decode("#ADD8E6");
    /**
     * The color <font color="#B0E0E6"> PowderBlue </font>
     **/
    public static final Color PowderBlue      = Color.decode("#B0E0E6");
    /**
     * The color <font color="#AFEEEE"> PaleTurquoise </font>
     **/
    public static final Color PaleTurquoise   = Color.decode("#AFEEEE");
    /**
     * The color <font color="#E0FFFF"> LightCyan </font>
     **/
    public static final Color LightCyan       = Color.decode("#E0FFFF");
    /**
     * The color <font color="#F0F8FF"> AliceBlue </font>
     **/
    public static final Color AliceBlue       = Color.decode("#F0F8FF");
    /**
     * The color <font color="#F0FFFF"> Azure </font>
     **/
    public static final Color Azure           = Color.decode("#F0FFFF");
  }
  public static class Browns
  {
    /**
     * The color <font color="#800000"> Maroon </font>
     **/
    public static final Color Maroon         = Color.decode("#800000");
    /**
     * The color <font color="#8B4513"> SaddleBrown </font>
     **/
    public static final Color SaddleBrown    = Color.decode("#8B4513");
    /**
     * The color <font color="#A52A2A"> Brown </font>
     **/
    public static final Color Brown          = Color.decode("#A52A2A");
    /**
     * The color <font color="#A0522D"> Sienna </font>
     **/
    public static final Color Sienna         = Color.decode("#A0522D");
    /**
     * The color <font color="#B8860B"> DarkGoldenrod </font>
     **/
    public static final Color DarkGoldenrod  = Color.decode("#B8860B");
    /**
     * The color <font color="#D2691E"> Chocolate </font>
     **/
    public static final Color Chocolate      = Color.decode("#D2691E");
    /**
     * The color <font color="#CD853F"> Peru </font>
     **/
    public static final Color Peru           = Color.decode("#CD853F");
    /**
     * The color <font color="#DAA520"> Goldenrod </font>
     **/
    public static final Color Goldenrod      = Color.decode("#DAA520");
    /**
     * The color <font color="#BC8F8F"> RosyBrown </font>
     **/
    public static final Color RosyBrown      = Color.decode("#BC8F8F");
    /**
     * The color <font color="#F4A460"> SandyBrown </font>
     **/
    public static final Color SandyBrown     = Color.decode("#F4A460");
    /**
     * The color <font color="#D2B48C"> Tan </font>
     **/
    public static final Color Tan            = Color.decode("#D2B48C");
    /**
     * The color <font color="#DEB887"> BurlyWood </font>
     **/
    public static final Color BurlyWood      = Color.decode("#DEB887");
    /**
     * The color <font color="#F5DEB3"> Wheat </font>
     **/
    public static final Color Wheat          = Color.decode("#F5DEB3");
    /**
     * The color <font color="#FFDEAD"> NavajoWhite </font>
     **/
    public static final Color NavajoWhite    = Color.decode("#FFDEAD");
    /**
     * The color <font color="#FFE4C4"> Bisque </font>
     **/
    public static final Color Bisque         = Color.decode("#FFE4C4");
    /**
     * The color <font color="#FFEBCD"> BlanchedAlmond </font>
     **/
    public static final Color BlanchedAlmond = Color.decode("#FFEBCD");
    /**
     * The color <font color="#FFF8DC"> Cornsilk </font>
     **/
    public static final Color Cornsilk       = Color.decode("#FFF8DC");
  }
  public static class Grays
  {
    /**
     * The color <font color="#000000"> Black </font>
     **/
    public static final Color Black          = Color.decode("#000000");
    /**
     * The color <font color="#2F4F4F"> DarkSlateGray </font>
     **/
    public static final Color DarkSlateGray  = Color.decode("#2F4F4F");
    /**
     * The color <font color="#696969"> DimGray </font>
     **/
    public static final Color DimGray        = Color.decode("#696969");
    /**
     * The color <font color="#808080"> Gray </font>
     **/
    public static final Color Gray           = Color.decode("#808080");
    /**
     * The color <font color="#708090"> SlateGray </font>
     **/
    public static final Color SlateGray      = Color.decode("#708090");
    /**
     * The color <font color="#778899"> LightSlateGray </font>
     **/
    public static final Color LightSlateGray = Color.decode("#778899");
    /**
     * The color <font color="#A9A9A9"> DarkGray </font>
     **/
    public static final Color DarkGray       = Color.decode("#A9A9A9");
    /**
     * The color <font color="#C0C0C0"> Silver </font>
     **/
    public static final Color Silver         = Color.decode("#C0C0C0");
    /**
     * The color <font color="#D3D3D3"> LightGray </font>
     **/
    public static final Color LightGray      = Color.decode("#D3D3D3");
    /**
     * The color <font color="#DCDCDC"> Gainsboro </font>
     **/
    public static final Color Gainsboro      = Color.decode("#DCDCDC");
  }
  public static class Greens
  {
    /**
     * The color <font color="#006400"> DarkGreen </font>
     **/
    public static final Color DarkGreen         = Color.decode("#006400");
    /**
     * The color <font color="#008000"> Green </font>
     **/
    public static final Color Green             = Color.decode("#008000");
    /**
     * The color <font color="#228B22"> ForestGreen </font>
     **/
    public static final Color ForestGreen       = Color.decode("#228B22");
    /**
     * The color <font color="#556B2F"> DarkOliveGreen </font>
     **/
    public static final Color DarkOliveGreen    = Color.decode("#556B2F");
    /**
     * The color <font color="#00FF00"> Lime </font>
     **/
    public static final Color Lime              = Color.decode("#00FF00");
    /**
     * The color <font color="#808000"> Olive </font>
     **/
    public static final Color Olive             = Color.decode("#808000");
    /**
     * The color <font color="#008080"> Teal </font>
     **/
    public static final Color Teal              = Color.decode("#008080");
    /**
     * The color <font color="#2E8B57"> SeaGreen </font>
     **/
    public static final Color SeaGreen          = Color.decode("#2E8B57");
    /**
     * The color <font color="#008B8B"> DarkCyan </font>
     **/
    public static final Color DarkCyan          = Color.decode("#008B8B");
    /**
     * The color <font color="#6B8E23"> OliveDrab </font>
     **/
    public static final Color OliveDrab         = Color.decode("#6B8E23");
    /**
     * The color <font color="#32CD32"> LimeGreen </font>
     **/
    public static final Color LimeGreen         = Color.decode("#32CD32");
    /**
     * The color <font color="#3CB371"> MediumSeaGreen </font>
     **/
    public static final Color MediumSeaGreen    = Color.decode("#3CB371");
    /**
     * The color <font color="#7CFC00"> LawnGreen </font>
     **/
    public static final Color LawnGreen         = Color.decode("#7CFC00");
    /**
     * The color <font color="#20B2AA"> LightSeaGreen </font>
     **/
    public static final Color LightSeaGreen     = Color.decode("#20B2AA");
    /**
     * The color <font color="#7FFF00"> Chartreuse </font>
     **/
    public static final Color Chartreuse        = Color.decode("#7FFF00");
    /**
     * The color <font color="#00FF7F"> SpringGreen </font>
     **/
    public static final Color SpringGreen       = Color.decode("#00FF7F");
    /**
     * The color <font color="#00FA9A"> MediumSpringGreen </font>
     **/
    public static final Color MediumSpringGreen = Color.decode("#00FA9A");
    /**
     * The color <font color="#9ACD32"> YellowGreen </font>
     **/
    public static final Color YellowGreen       = Color.decode("#9ACD32");
    /**
     * The color <font color="#00CED1"> DarkTurquoise </font>
     **/
    public static final Color DarkTurquoise     = Color.decode("#00CED1");
    /**
     * The color <font color="#8FBC8F"> DarkSeaGreen </font>
     **/
    public static final Color DarkSeaGreen      = Color.decode("#8FBC8F");
    /**
     * The color <font color="#ADFF2F"> GreenYellow </font>
     **/
    public static final Color GreenYellow       = Color.decode("#ADFF2F");
    /**
     * The color <font color="#66CDAA"> MediumAquamarine </font>
     **/
    public static final Color MediumAquamarine  = Color.decode("#66CDAA");
    /**
     * The color <font color="#48D1CC"> MediumTurquoise </font>
     **/
    public static final Color MediumTurquoise   = Color.decode("#48D1CC");
    /**
     * The color <font color="#40E0D0"> Turquoise </font>
     **/
    public static final Color Turquoise         = Color.decode("#40E0D0");
    /**
     * The color <font color="#90EE90"> LightGreen </font>
     **/
    public static final Color LightGreen        = Color.decode("#90EE90");
    /**
     * The color <font color="#98FB98"> PaleGreen </font>
     **/
    public static final Color PaleGreen         = Color.decode("#98FB98");
    /**
     * The color <font color="#7FFFD4"> Aquamarine </font>
     **/
    public static final Color Aquamarine        = Color.decode("#7FFFD4");
    /**
     * The color <font color="#AFEEEE"> PaleTurquoise </font>
     **/
    public static final Color PaleTurquoise     = Color.decode("#AFEEEE");
  }
  public static class Oranges
  {
    /**
     * The color <font color="#FF4500"> OrangeRed </font>
     **/
    public static final Color OrangeRed   = Color.decode("#FF4500");
    /**
     * The color <font color="#FF8C00"> DarkOrange </font>
     **/
    public static final Color DarkOrange  = Color.decode("#FF8C00");
    /**
     * The color <font color="#FFA500"> Orange </font>
     **/
    public static final Color Orange      = Color.decode("#FFA500");
    /**
     * The color <font color="#FF6347"> Tomato </font>
     **/
    public static final Color Tomato      = Color.decode("#FF6347");
    /**
     * The color <font color="#FF7F50"> Coral </font>
     **/
    public static final Color Coral       = Color.decode("#FF7F50");
    /**
     * The color <font color="#FFA07A"> LightSalmon </font>
     **/
    public static final Color LightSalmon = Color.decode("#FFA07A");
  }
  public static class Pinks
  {
    /**
     * The color <font color="#C71585"> MediumVioletRed </font>
     **/
    public static final Color MediumVioletRed = Color.decode("#C71585");
    /**
     * The color <font color="#FF1493"> DeepPink </font>
     **/
    public static final Color DeepPink        = Color.decode("#FF1493");
    /**
     * The color <font color="#DB7093"> PaleVioletRed </font>
     **/
    public static final Color PaleVioletRed   = Color.decode("#DB7093");
    /**
     * The color <font color="#FF00FF"> Fuchsia </font>
     **/
    public static final Color Fuchsia         = Color.decode("#FF00FF");
    /**
     * The color <font color="#FF69B4"> HotPink </font>
     **/
    public static final Color HotPink         = Color.decode("#FF69B4");
    /**
     * The color <font color="#FFB6C1"> LightPink </font>
     **/
    public static final Color LightPink       = Color.decode("#FFB6C1");
    /**
     * The color <font color="#FFC0CB"> Pink </font>
     **/
    public static final Color Pink            = Color.decode("#FFC0CB");
  }
  public static class Purples
  {
    /**
     * The color <font color="#4B0082"> Indigo </font>
     **/
    public static final Color Indigo          = Color.decode("#4B0082");
    /**
     * The color <font color="#800080"> Purple </font>
     **/
    public static final Color Purple          = Color.decode("#800080");
    /**
     * The color <font color="#483D8B"> DarkSlateBlue </font>
     **/
    public static final Color DarkSlateBlue   = Color.decode("#483D8B");
    /**
     * The color <font color="#8B008B"> DarkMagenta </font>
     **/
    public static final Color DarkMagenta     = Color.decode("#8B008B");
    /**
     * The color <font color="#C71585"> MediumVioletRed </font>
     **/
    public static final Color MediumVioletRed = Color.decode("#C71585");
    /**
     * The color <font color="#9400D3"> DarkViolet </font>
     **/
    public static final Color DarkViolet      = Color.decode("#9400D3");
    /**
     * The color <font color="#6A5ACD"> SlateBlue </font>
     **/
    public static final Color SlateBlue       = Color.decode("#6A5ACD");
    /**
     * The color <font color="#8A2BE2"> BlueViolet </font>
     **/
    public static final Color BlueViolet      = Color.decode("#8A2BE2");
    /**
     * The color <font color="#9932CC"> DarkOrchid </font>
     **/
    public static final Color DarkOrchid      = Color.decode("#9932CC");
    /**
     * The color <font color="#7B68EE"> MediumSlateBlue </font>
     **/
    public static final Color MediumSlateBlue = Color.decode("#7B68EE");
    /**
     * The color <font color="#BC8F8F"> RosyBrown </font>
     **/
    public static final Color RosyBrown       = Color.decode("#BC8F8F");
    /**
     * The color <font color="#9370DB"> MediumPurple </font>
     **/
    public static final Color MediumPurple    = Color.decode("#9370DB");
    /**
     * The color <font color="#DB7093"> PaleVioletRed </font>
     **/
    public static final Color PaleVioletRed   = Color.decode("#DB7093");
    /**
     * The color <font color="#BA55D3"> MediumOrchid </font>
     **/
    public static final Color MediumOrchid    = Color.decode("#BA55D3");
    /**
     * The color <font color="#FF00FF"> Fuchsia </font>
     **/
    public static final Color Fuchsia         = Color.decode("#FF00FF");
    /**
     * The color <font color="#FF00FF"> Magenta </font>
     **/
    public static final Color Magenta         = Color.decode("#FF00FF");
    /**
     * The color <font color="#DA70D6"> Orchid </font>
     **/
    public static final Color Orchid          = Color.decode("#DA70D6");
    /**
     * The color <font color="#DDA0DD"> Plum </font>
     **/
    public static final Color Plum            = Color.decode("#DDA0DD");
    /**
     * The color <font color="#EE82EE"> Violet </font>
     **/
    public static final Color Violet          = Color.decode("#EE82EE");
    /**
     * The color <font color="#D8BFD8"> Thistle </font>
     **/
    public static final Color Thistle         = Color.decode("#D8BFD8");
    /**
     * The color <font color="#E6E6FA"> Lavender </font>
     **/
    public static final Color Lavender        = Color.decode("#E6E6FA");
  }
  public static class Reds
  {
    /**
     * The color <font color="#8B0000"> DarkRed </font>
     **/
    public static final Color DarkRed         = Color.decode("#8B0000");
    /**
     * The color <font color="#B22222"> FireBrick </font>
     **/
    public static final Color FireBrick       = Color.decode("#B22222");
    /**
     * The color <font color="#FF0000"> Red </font>
     **/
    public static final Color Red             = Color.decode("#FF0000");
    /**
     * The color <font color="#DC143C"> Crimson </font>
     **/
    public static final Color Crimson         = Color.decode("#DC143C");
    /**
     * The color <font color="#FF4500"> OrangeRed </font>
     **/
    public static final Color OrangeRed       = Color.decode("#FF4500");
    /**
     * The color <font color="#C71585"> MediumVioletRed </font>
     **/
    public static final Color MediumVioletRed = Color.decode("#C71585");
    /**
     * The color <font color="#CD5C5C"> IndianRed </font>
     **/
    public static final Color IndianRed       = Color.decode("#CD5C5C");
    /**
     * The color <font color="#FF6347"> Tomato </font>
     **/
    public static final Color Tomato          = Color.decode("#FF6347");
    /**
     * The color <font color="#DB7093"> PaleVioletRed </font>
     **/
    public static final Color PaleVioletRed   = Color.decode("#DB7093");
    /**
     * The color <font color="#FA8072"> Salmon </font>
     **/
    public static final Color Salmon          = Color.decode("#FA8072");
    /**
     * The color <font color="#F08080"> LightCoral </font>
     **/
    public static final Color LightCoral      = Color.decode("#F08080");
    /**
     * The color <font color="#E9967A"> DarkSalmon </font>
     **/
    public static final Color DarkSalmon      = Color.decode("#E9967A");
    /**
     * The color <font color="#FFA07A"> LightSalmon </font>
     **/
    public static final Color LightSalmon     = Color.decode("#FFA07A");
  }
  public static class Whites
  {
    /**
     * The color <font color="#FAEBD7"> AntiqueWhite </font>
     **/
    public static final Color AntiqueWhite  = Color.decode("#FAEBD7");
    /**
     * The color <font color="#FFE4E1"> MistyRose </font>
     **/
    public static final Color MistyRose     = Color.decode("#FFE4E1");
    /**
     * The color <font color="#F5F5DC"> Beige </font>
     **/
    public static final Color Beige         = Color.decode("#F5F5DC");
    /**
     * The color <font color="#FAF0E6"> Linen </font>
     **/
    public static final Color Linen         = Color.decode("#FAF0E6");
    /**
     * The color <font color="#FDF5E6"> OldLace </font>
     **/
    public static final Color OldLace       = Color.decode("#FDF5E6");
    /**
     * The color <font color="#F0FFF0"> Honeydew </font>
     **/
    public static final Color Honeydew      = Color.decode("#F0FFF0");
    /**
     * The color <font color="#F5F5F5"> WhiteSmoke </font>
     **/
    public static final Color WhiteSmoke    = Color.decode("#F5F5F5");
    /**
     * The color <font color="#FFF5EE"> Seashell </font>
     **/
    public static final Color Seashell      = Color.decode("#FFF5EE");
    /**
     * The color <font color="#FFF0F5"> LavenderBlush </font>
     **/
    public static final Color LavenderBlush = Color.decode("#FFF0F5");
    /**
     * The color <font color="#F0F8FF"> AliceBlue </font>
     **/
    public static final Color AliceBlue     = Color.decode("#F0F8FF");
    /**
     * The color <font color="#FFFAF0"> FloralWhite </font>
     **/
    public static final Color FloralWhite   = Color.decode("#FFFAF0");
    /**
     * The color <font color="#F5FFFA"> MintCream </font>
     **/
    public static final Color MintCream     = Color.decode("#F5FFFA");
    /**
     * The color <font color="#F0FFFF"> Azure </font>
     **/
    public static final Color Azure         = Color.decode("#F0FFFF");
    /**
     * The color <font color="#FFFFF0"> Ivory </font>
     **/
    public static final Color Ivory         = Color.decode("#FFFFF0");
    /**
     * The color <font color="#F8F8FF"> GhostWhite </font>
     **/
    public static final Color GhostWhite    = Color.decode("#F8F8FF");
    /**
     * The color <font color="#FFFAFA"> Snow </font>
     **/
    public static final Color Snow          = Color.decode("#FFFAFA");
    /**
     * The color <font color="#FFFFFF"> White </font>
     **/
    public static final Color White         = Color.decode("#FFFFFF");
  }
  public static class Yellows
  {
    /**
     * The color <font color="#B8860B"> DarkGoldenrod </font>
     **/
    public static final Color DarkGoldenrod        = Color.decode("#B8860B");
    /**
     * The color <font color="#DAA520"> Goldenrod </font>
     **/
    public static final Color Goldenrod            = Color.decode("#DAA520");
    /**
     * The color <font color="#FFD700"> Gold </font>
     **/
    public static final Color Gold                 = Color.decode("#FFD700");
    /**
     * The color <font color="#BDB76B"> DarkKhaki </font>
     **/
    public static final Color DarkKhaki            = Color.decode("#BDB76B");
    /**
     * The color <font color="#FFFF00"> Yellow </font>
     **/
    public static final Color Yellow               = Color.decode("#FFFF00");
    /**
     * The color <font color="#F0E68C"> Khaki </font>
     **/
    public static final Color Khaki                = Color.decode("#F0E68C");
    /**
     * The color <font color="#EEE8AA"> PaleGoldenrod </font>
     **/
    public static final Color PaleGoldenrod        = Color.decode("#EEE8AA");
    /**
     * The color <font color="#FFDAB9"> PeachPuff </font>
     **/
    public static final Color PeachPuff            = Color.decode("#FFDAB9");
    /**
     * The color <font color="#FFE4B5"> Moccasin </font>
     **/
    public static final Color Moccasin             = Color.decode("#FFE4B5");
    /**
     * The color <font color="#FFEFD5"> PapayaWhip </font>
     **/
    public static final Color PapayaWhip           = Color.decode("#FFEFD5");
    /**
     * The color <font color="#FFFACD"> LemonChiffon </font>
     **/
    public static final Color LemonChiffon         = Color.decode("#FFFACD");
    /**
     * The color <font color="#FAFAD2"> LightGoldenrodYellow </font>
     **/
    public static final Color LightGoldenrodYellow = Color.decode("#FAFAD2");
    /**
     * The color <font color="#FFFFE0"> LightYellow </font>
     **/
    public static final Color LightYellow          = Color.decode("#FFFFE0");
  }
  private static Random RANDOM = new Random();
  public static Color darken(Color color)
  {
    int red = decrease(color.getRed(), 10);
    int green = decrease(color.getGreen(), 10);
    int blue = decrease(color.getBlue(), 10);
    return new Color(red, green, blue);
  }
  private static int decrease(int amount, int decreaseBy)
  {
    return Math.max(0, amount - decreaseBy);
  }
  public static Color getRandomColor()
  {
    return new Color(RANDOM.nextInt(255), RANDOM.nextInt(255), RANDOM.nextInt(255));
  }
  public static void mockRandom()
  {
    RANDOM.setSeed(5);
  }
  public static int calculateTransparency(int percentTransparent)
  {
    int opacity = (percentTransparent * 255) / 100;
    return (255 - opacity);
  }
  public static Color getTransparentVersion(Color color, int percentTransparency)
  {
    return new Color(color.getRed(), color.getGreen(), color.getBlue(),
        calculateTransparency(percentTransparency));
  }
}