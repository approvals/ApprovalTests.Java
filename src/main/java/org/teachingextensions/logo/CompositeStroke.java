package org.teachingextensions.logo;

import java.awt.Shape;
import java.awt.Stroke;

public class CompositeStroke implements Stroke
{
  private Stroke stroke1, stroke2;
  public CompositeStroke(Stroke stroke1, Stroke stroke2)
  {
    this.stroke1 = stroke1;
    this.stroke2 = stroke2;
  }
  public Shape createStrokedShape(Shape shape)
  {
    return stroke2.createStrokedShape(stroke1.createStrokedShape(shape));
  }
}
