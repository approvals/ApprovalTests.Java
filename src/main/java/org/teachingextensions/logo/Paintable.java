package org.teachingextensions.logo;

import java.awt.Graphics2D;

public interface Paintable
{
  public static Paintable NOTHING = new Paintable()
                                  {
                                    public void paint(Graphics2D g)
                                    {
                                    }
                                  };
  void paint(Graphics2D g);
}
