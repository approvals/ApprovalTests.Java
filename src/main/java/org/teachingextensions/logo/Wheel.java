package org.teachingextensions.logo;

import java.util.ArrayList;
import java.util.List;

import com.spun.util.NumberUtils;

/**
 * A Wheel is a List with no ending on beginning <br/>
 * 
 * <b>Example:</b> {@code  
 *    Wheel<String> names = new Wheel<String>();  
 *    names.add("Chocolate"); 
 *    names.add("Peanut Butter"); 
 *    for(int i = 0; i < 6; i++)  > {
 * String name = names.next();
 * System.out.print(name + " ");
 * }
 * }
 * Would result in : <br/>
 * Chocolate Peanut Butter Chocolate Peanut Butter Chocolate Peanut Butter
 * 
 * @param <T>
 */
public class Wheel<T>
{
  private List<T> list  = new ArrayList<T>();
  private int     index = 0;
  public Wheel(T... loadWith)
  {
    for (T t : loadWith)
    {
      add(t);
    }
  }
  public Wheel()
  {
  }
  public void add(T i)
  {
    list.add(i);
  }
  public T next()
  {
    assertNonEmpty();
    if (index >= list.size())
    {
      index = 0;
    }
    T t = list.get(index++);
    return t;
  }
  private void assertNonEmpty()
  {
    if (list.isEmpty())
    {
      String message = "I call shenanigans!!!\nThis Wheel is empty\nYou can NOT get something from the Wheel before you've added anything to it.";
      throw new RuntimeException(message);
    }
  }
  public T getRandomFrom()
  {
    assertNonEmpty();
    int index = NumberUtils.getRandomInt(0, list.size());
    return list.get(index);
  }
  public void empty()
  {
    list.clear();
    index = 0;
  }
}
