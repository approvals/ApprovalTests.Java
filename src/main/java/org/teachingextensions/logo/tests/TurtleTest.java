package org.teachingextensions.logo.tests;

import java.awt.Color;

import junit.framework.TestCase;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.DelayedClipboardReporter;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.lambda.functions.Function1;
import org.teachingextensions.logo.Colors;
import org.teachingextensions.logo.Turtle;
import org.teachingextensions.logo.Wheel;

@UseReporter({DiffReporter.class, DelayedClipboardReporter.class})
public class TurtleTest extends TestCase
{
  public void testShow() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    Approvals.verify(turtle.getImage());
  }
  public void testRotate() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    turtle.turn(90);
    Approvals.verify(turtle.getImage());
  }
  public void testRotate45() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    turtle.turn(45);
    Approvals.verify(turtle.getImage());
    assertEquals(45.0, turtle.getAngleInDegrees(), 0.02);
  }
  public void testRotate45CounterClockwise() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    turtle.turn(-45);
    Approvals.verify(turtle.getImage());
  }
  public void testMove() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    turtle.move(50);
    Approvals.verify(turtle.getImage());
  }
  public void testMove45DegreesLeft() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    turtle.turn(-45);
    turtle.move(50);
    Approvals.verify(turtle.getImage());
  }
  public void testDelta() throws Exception
  {
    String out = "";
    for (int i = 0; i < 9; i++)
    {
      double deltaY = Turtle.getDeltaY(5, i * 45.0);
      double deltaX = Turtle.getDeltaX(5, i * 45.0);
      out += String.format("%s=>[%s, %s]\r\n", i * 45, deltaX, deltaY);
    }
    Approvals.verify(out);
  }
  public void testCalculateAngle() throws Exception
  {
    assertEquals(-90.0, Turtle.angleCalculator(0, 0, -1, 0), 0.005);
    assertEquals(90.0, Turtle.angleCalculator(0, 0, 1, 0), 0.005);
  }
  public void testPentagonCrazy() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    Wheel<Color> wheel = new Wheel<Color>();
    wheel.add(Colors.Blues.SteelBlue);
    wheel.add(Colors.Purples.DarkMagenta);
    wheel.add(Colors.Blues.DarkSlateBlue);
    wheel.add(Colors.Purples.Indigo);
    wheel.add(Colors.Purples.DarkOrchid);
    Wheel<Integer> thickness = new Wheel<Integer>(1, 2, 3);
    for (int i = 0; i < 200; i++)
    {
      turtle.setPenColor(wheel.next());
      turtle.setPenWidth(thickness.next());
      turtle.move(i + 1);
      turtle.turn(360 / 5);
      turtle.turn(1);
    }
    Approvals.verify(turtle.getImage());
  }
  public void testPenDown() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    for (int i = 0; i < 3; i++)
    {
      turtle.move(10);
      turtle.penUp();
      turtle.move(10);
      turtle.penDown();
    }
    Approvals.verify(turtle.getImage());
  }
  public void testHide() throws Exception
  {
    Turtle turtle = TurtleUtils.getTurtle();
    turtle.move(50);
    turtle.hide();
    Approvals.verify(turtle.getImage());
  }
  public void testSpeed() throws Exception
  {
    Integer[] speeds = {-5, 5, 15, Turtle.TEST_SPEED};
    Approvals.verifyAll("Speeds", speeds, new Function1<Integer, String>()
    {
      @Override
      public String call(Integer speed)
      {
        try
        {
          Turtle turtle = TurtleUtils.getTurtle();
          turtle.setSpeed(speed);
          return speed + " => " + turtle.getSpeed();
        }
        catch (Exception e)
        {
          return speed + " => " + e.getMessage();
        }
      }
    });
  }
}
