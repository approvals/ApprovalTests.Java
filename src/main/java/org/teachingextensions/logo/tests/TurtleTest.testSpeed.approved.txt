Speeds


-5 => I call shenanigans!!!
The speed '-5' is not between the acceptable range of [1-10]
Perhaps you should read the documentation
5 => 5
15 => I call shenanigans!!!
The speed '15' is not between the acceptable range of [1-10]
Perhaps you should read the documentation
-2147483648 => -2147483648
